sdk-api-kotlin/src/main/kotlin/dev/restate/sdk/kotlin/api.kt

```diff
@@ -294,13 +294,51 @@ sealed interface ObjectContext : SharedObjectContext {
   suspend fun clearAll()
 }
 
+/**
+ * This interface extends [Context] adding access to the workflow instance key-value state storage
+ * and to the [DurablePromise] API.
+ *
+ * This interface can be used only within shared handlers of workflow classes/interfaces.
+ *
+ * NOTE: This interface MUST NOT be accessed concurrently since it can lead to different orderings
+ * of user actions, corrupting the execution of the invocation.
+ *
+ * @see Context
+ * @see SharedObjectContext
+ */
 sealed interface SharedWorkflowContext : SharedObjectContext {
-  fun <T : Any> durablePromise(key: DurablePromiseKey<T>): DurablePromise<T>
+  /**
+   * Create a [DurablePromise] for the given key.
+   *
+   * You can use this feature to implement interaction between different workflow handlers, e.g. to
+   * send a signal from a shared handler to the workflow handler.
+   *
+   * @see DurablePromise
+   */
+  fun <T : Any> promise(key: DurablePromiseKey<T>): DurablePromise<T>
 
-  fun <T : Any> durablePromiseHandle(key: DurablePromiseKey<T>): DurablePromiseHandle<T>
+  /**
+   * Create a new [DurablePromiseHandle] for the provided key. You can use it to
+   * [DurablePromiseHandle.resolve] or [DurablePromiseHandle.reject] the given [DurablePromise].
+   *
+   * @see DurablePromise
+   */
+  fun <T : Any> promiseHandle(key: DurablePromiseKey<T>): DurablePromiseHandle<T>
 }
 
-sealed interface WorkflowContext : SharedWorkflowContext, ObjectContext {}
+/**
+ * This interface extends [Context] adding access to the workflow instance key-value state storage
+ * and to the [DurablePromise] API.
+ *
+ * This interface can be used only within workflow handlers of workflow classes/interfaces.
```

Comment:
Move this line up. First say what it is:
`This interface should be used as the context for the run handler of workflows.`